hbase-server/src/test/java/org/apache/hadoop/hbase/replication/regionserver/TestRefreshRecoveredReplication.java

```diff
@@ -75,6 +80,7 @@
 
   @BeforeClass
   public static void setUpBeforeClass() throws Exception {
+    // NUM_SLAVES1 is presumed 2 in below.
```

Comment:
Can `NUM_SLAVES1` be made final, and referenced wherever it's presumed to be 2? Maybe rename it if you're in there making that change...

Reply:
The NUM_SLAVES1 comes from base class.

This class and the subclasses need major refactoring. I avoid it in here trying to keep change minimal.